bluesky/plans.py

```diff
@@ -16,13 +18,41 @@
 from . import plan_patterns
 
 from . import utils
-from .utils import Msg, get_hinted_fields
+from .utils import (
+    Msg,
+    get_hinted_fields,
+    ScalarOrIterable,
+    CustomPlanMetadata,
+    MsgGenerator,
+)
 
 from . import preprocessors as bpp
 from . import plan_stubs as bps
-
-
-def count(detectors, num=1, delay=None, *, per_shot=None, md=None):
+from .protocols import Flyable, Movable, Readable
+
+#: Plan function that can be used for each shot in a detector acquisition involving no actuation
+PerShot = Callable[[Iterable[Readable], Optional[bps.TakeReading]], MsgGenerator]
+
+#: Plan function that can be used for each step in a scan
+PerStep = Callable[
+    [
+        Iterable[Readable],
+        Mapping[Movable, Any],
+        Mapping[Movable, Any],
+        Optional[bps.TakeReading],
+    ],
+    MsgGenerator,
+]
+
+
+def count(
+    detectors: List[Readable],
+    num: Optional[int] = 1,
```

Comment:
The type should probably be simply `int`. `Optional[int]` assumes that the parameter accepts `None`, which is not true here.

Reply:
Count does accept `num=None`, see line 84 (on this branch)
```python
    if num is None:
        num_intervals = None
```